v2/yarpcjson/codec_test.go

```diff
@@ -48,12 +49,13 @@ func TestNewCodec(t *testing.T) {
 
 func TestJsonCodec_Decode(t *testing.T) {
 	c := jsonCodec{
-		reader: mapReader{reflect.TypeOf(make(map[string]interface{}))},
+		reader: structReader{reflect.TypeOf(simpleResponse{})},
 	}
 
-	jsonContent := `{"foo": 42}`
+	jsonContent := `{"Success":true}`
 	validReqBuf := yarpc.NewBufferString(jsonContent)
-	_, err := c.Decode(validReqBuf)
+	body, err := c.Decode(validReqBuf)
+	assert.Equal(t, *body.(*simpleResponse), simpleResponse{Success: true})
```

Comment:
nit: I think we can remove the dereference of `body` and assert it against `&simpleResponse{Success: true}` here:
```
assert.Equal(t, body.(*simpleResponse), &simpleResponse{Success: true})
```

Reply:
Why do we prefer comparing the pointers over the base values?

Reply:
`assert.Equal` should do a deep equals for us, so the dereference is unnecessary. Either way is fine though.